### DISCUSSION TITLE
Creating a V6 custom Image Filter?

### DISCUSSION THREAD

**Discussion by VIZULR** [fabricjs/fabric.js#8959] [Q&A]
Has anyone had any luck getting custom image filters in V6 to work, I have some filters I'm trying to port from V5. I understand the boilerplate, but I just can't get any imports working from within my own folder of filters, for example how do I - import { BaseFilter } etc. Any help on the setup would be appreciated. ;)

**Comment by arifhussainfilestack** [comment]
I hope this message finds you well. I'm reaching out because I'm encountering an issue while working with Fabric.js v6 in TypeScript.
I've extended fabric.FabricImage.filters.BaseFilter to create a custom filter class called FabricLutFilter. 
export class FabricLutFilter extends fabric.FabricImage.filters.BaseFilter 
However, I'm receiving the following TypeScript error:
Property 'filters' does not exist on type 'typeof FabricImage'.ts(2339)
Could you please advise on how to resolve this issue? Any guidance or solution you could provide would be greatly appreciated.

**Comment by asturur** [comment]
https://github.com/fabricjs/fabricjs.github.io/blob/main/src/content/demo/custom-filter-swap-color/colorSwapFilter.ts

This is a custom filter made for the website with the v6 codebase. Let me know if it helps

**Comment by bo-zn** [comment]
https://github.com/bo-zn/image-editor/blob/main/src/filters/exposure.ts     
    
I used v6 to make a custom filter about image exposure, I hope it can help you
